.travis.yml

```diff
@@ -45,6 +63,16 @@ matrix:
   #          CONDA_DEPENDENCIES='sphinx cython numpy pytest-cov'
   #          EVENT_TYPE='push pull_request cron'
 
+before_install:
+
+      # Test OSX without OpenMP support
+      # Since the matrix OSX tests use the OS shipped version of clang, they double up
+      # as exploratory tests for when the shipped version has automatic OpenMP support.
+      # These tests will then fail and at such a time a new one should be added
+      # to explicitly remove OpenMP support.
+    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export TEST_OPENMP="False";
+      else export TEST_OPENMP="True"; fi
```

Comment:
Remove the `before_install` section and define the env variables in the jobs above.

Reply:
I'll double check this, but if I remember correctly, it needs to be here so as to also work with the matrix permutations. There is no way to add this to them as they will be evaluated not at the correct stage, i.e.  TRAVIS_OS_NAME is undefined. For the explicit tests it is, which is why I do so there also, but it isn't for the `env:matrix:` sec.

Reply:
Regardless of OS, ``TEST_OPENMP`` has to be set to something. If set as a global env and to ``true`` it will prevent the OSX tests from passing. If set to ``false`` it will prevent the linux tests from passing (unless we explicitly turn off the openmp support - which is altogether silly). One alternative is to make it such that all global matrixed OS tests pass with either state, which probably means getting OpenMP to work for _all_ the OSX tests. The other alternative is to make  ``TEST_OPENMP`` conditional and this therefore can't happen in the global env stage by definition. It has to go in the ``install`` or better still (due to overloading and its current use, not to mention semantically) in the ``before_install`` stage. Let me know if there is an alternative, perhaps the entire idea needs to be re-thought, however, I honestly don't see the cause to _not_ use the ``before_install_ stage.